zp-relayer/utils/constants.ts

```diff
@@ -15,6 +15,8 @@ const constants = {
   POOL_ID_SIZE: 6,
   ZERO_ADDRESS: '0x0000000000000000000000000000000000000000',
   INIT_ROOT: '11469701942666298368112882412133877458305516134926649826543144744382391691533',
+  BASE_CALLDATA_SIZE: 644, // constant calldata size for tx without memo
+  OP_GAS_ORACLE_ADDRESS: '0x420000000000000000000000000000000000000F',
```

Comment:
Are you sure it's a good idea to define price oracle contract address for a concrete chain as a constant? Maybe it's better to set it as a configuration variable?

Reply:
This is a predefined OP oracle contract deployed at genesis block, so it is not supposed to change in the future. I don't think we should make it a variable parameter, as it will "always" be the same.

Reply:
Will this address be the same for other OP-stack based chains, e.g. Base?

Reply:
Just checked it on goerli testnet, and looks like they use the same address for gas price oracle. The difference, is that it is deployed using a proxy, but this shouldn't matter.
https://goerli.basescan.org/address/0x420000000000000000000000000000000000000f#code

Reply:
>You made separated class and constant for Optimism only. It doesn't look like universal solution

I don't think there is a universal solution for all chains even with a similar fee charging scheme. E.g. for Arbitrum (also an optimistic rollup) we will need a separate class with different logic because they use different contracts with different ABIs . For zkSync the difference will be more significant.

https://developer.arbitrum.io/arbos/l1-pricing